### ISSUE TITLE
Field with emptyList isn't decoded by ProtoBuf

### ISSUE THREAD

**Issue by SerVB** [Kotlin/kotlinx.serialization#806]
**To Reproduce**
If I do:

```kotlin
@Serializable
data class A(val myList: List<String>)

object Main {

    @JvmStatic
    fun main(vararg args: String) {
        val protobuf = ProtoBuf()

        val a = A(emptyList())

        val encoded = protobuf.dump(A.serializer(), a)
        val decoded = protobuf.load(A.serializer(), encoded)

        println(decoded == a)
    }
}
```

I get the following in console:

```
Exception in thread "main" kotlinx.serialization.MissingFieldException: Field 'myList' is required, but it was missing
	at A.<init>(Main.kt)
	at A$$serializer.deserialize(Main.kt)
	at A$$serializer.deserialize(Main.kt:59)
	at kotlinx.serialization.protobuf.ProtoBuf$ProtobufReader.decodeSerializableValue(ProtoBuf.kt:290)
	at kotlinx.serialization.DecodingKt.decode(Decoding.kt:521)
	at kotlinx.serialization.protobuf.ProtoBuf.load(ProtoBuf.kt:548)
	at Main.main(Main.kt:71)
```

**Expected behavior**

It should output "true" to the console.

**Workaround**

As workaround, I use `data class A(val myList: List<String> = emptyList())`.

**Environment**
 - Kotlin version: 1.3.71
 - Library version: 0.20.0
 - Kotlin platforms: JVM (JS is also checked with similar code)

**Comment by SerVB** [comment]
By the way, I've just seen the following in the [docs](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/runtime_usage.md#protobuf):

> Because format spec says that if the list is empty, there will be no elements in the stream with such tag, you must explicitly mark any field of list type with default `= emptyList()`. Same for maps.

I think it should be done automatically.

**Comment by qwwdfsad** [comment]
The compiler generates code in a general form for an abstract format and does not implies anything protobuf-specific. Adding an empty collection instead of a list is not a generally-applicable change.
Maybe we can make this change in the protobuf format itself, but we are not sure yet

**Comment by SerVB** [comment]
I think possibility of runtime exceptions is a good enough reason to do something with this.

**Comment by qwwdfsad** [comment]
The current solution is to fail-fast and to add an exception message that suggests adding an empty default value when attempting to serialize such value. It should prevent most of the bugs during the initial testing.

Automatically add empty collections is a really complicated topic and I am not sure it is possible to address completely, though we will try to it

**Comment by sandwwraith** [comment]
Related: #438

**Comment by MaryWeeb** [comment]
Is there an ETA on a fix for this?

**Comment by qwwdfsad** [comment]
Not yet, unfortunately

**Comment by sanity** [comment]
Ran into this problem too (inability to deserialize property with empty list), suggest a more informative exception?

**Comment by qwwdfsad** [comment]
Fixed in 1.2.2 with https://github.com/Kotlin/kotlinx.serialization/commit/8f82c6be67a6ceb9518a0557ea40bd7d82a20ab3
